datastore/gcloud/aio/datastore/value.py

```diff
@@ -35,6 +36,8 @@ def from_repr(cls, data: Dict[str, Any]) -> 'Value':
             if json_key in data:
                 if json_key == 'nullValue':
                     value = None
+                elif json_key == 'arrayValue':
```

Comment:
Looks like you've added this to `TypeName`, so you should be able to change this check to `elif value_type == Array:` (which would be great, since it would help us avoid copying that string around outside of our constants).